src/PostgREST/Plan.hs

```diff
@@ -1088,6 +1088,7 @@ negotiateContent conf ApiRequest{iAction=act, iPreferences=Preferences{preferRep
     -- TODO: despite no aggregate, these are responding with a Content-Type, which is not correct.
     (ActDb (ActRelationRead _ True),             Just (_, mt)) -> Right (NoAgg, mt)
     (ActDb (ActRoutine  _ (InvRead True)), Just (_, mt))             -> Right (NoAgg, mt)
+    (_, Just (_, MTApplicationJSONSQL))                          -> Right (NoAgg, MTApplicationJSONSQL)
```

Comment:
This media type should be opt in, I suggest reusing the https://docs.postgrest.org/en/v12/references/configuration.html#db-plan-enabled (`configDbPlanEnabled`) for this. See how it's done on:

https://github.com/PostgREST/postgrest/blob/c732591f37533171fd5bfd4d5c2566f578c03d84/src/PostgREST/Plan.hs#L1098-L1104

It's a bit messy now unfortunately.

Reply:
Ah so in a way on the Plan there should be a check to allow it if the `configDbPlanEnabled`? Honestly I totally understand that, enabling this does open up lots of internal details